packages/utils/src/misc.ts

```diff
@@ -25,13 +25,19 @@ export function uuid4(): string {
   const gbl = GLOBAL_OBJ as typeof GLOBAL_OBJ & CryptoGlobal;
   const crypto = gbl.crypto || gbl.msCrypto;
 
-  if (crypto && crypto.randomUUID) {
-    return crypto.randomUUID().replace(/-/g, '');
+  let getRandomByte = (): number => Math.random() * 16;
+  try {
+    if (crypto) {
+      if (crypto.randomUUID) {
+        return crypto.randomUUID().replace(/-/g, '');
+      }
+      getRandomByte = () => crypto.getRandomValues && crypto.getRandomValues(new Uint8Array(1))[0];
```

Comment:
m: Shouldn't we also guard this instead of checking this inside here? Otherwise, could this not happen to return `undefined`? (TS does not catch this because from it's POV `getRandomValues` is always present). So rather:

```js
if (crypto.randomUUID) {
  return crypto.randomUUID().replace(/-/g, '');
}
if (crypto.getRandomValues) {
    getRandomByte = () => crypto.getRandomValues(new Uint8Array(1))[0];
}
```

?

Reply:
ah, already resolved :D 🚀